x-pack/plugins/ingest_manager/server/routes/setup/handlers.ts

```diff
@@ -41,7 +43,26 @@ export const createFleetSetupHandler: RequestHandler<
       username: request.body.admin_username,
       password: request.body.admin_password,
     });
-    await agentConfigService.ensureDefaultAgentConfig(soClient);
+    await generateEnrollmentAPIKey(soClient, {
+      name: 'Default',
+      configId: await agentConfigService.getDefaultAgentConfigId(soClient),
+    });
+
+    return response.ok({
+      body: { isInitialized: true },
+    });
+  } catch (e) {
+    return response.customError({
+      statusCode: 500,
+      body: { message: e.message },
+    });
+  }
+};
+
+export const ingestManagerSetupHandler: RequestHandler = async (context, request, response) => {
+  const soClient = context.core.savedObjects.client;
+  try {
+    await setup(soClient);
```

Comment:
ill need to pass context.core.elasticsearch.adminClient.callAsCurrentUser to setup but I can do that when I add the install packages functionality